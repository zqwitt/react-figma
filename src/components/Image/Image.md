## Image

Analog of the `Image` component at [React Native](https://facebook.github.io/react-native/docs/image), 
[React Sketchapp](http://airbnb.io/react-sketchapp/docs/API.html#image). 
Renders Figma rectangle node with background.

#### Props

| Prop       | Type     | Default | Note                                              |
| ---------- | -------- | ------- | ------------------------------------------------- |
| `name`     | `String` |         | The name to be displayed in the Figma Layers List |
| `source`   | `String` |         | Path for loading image                            |
| `resizeMode`| `ResizeMode` |         |                                              |
| `style`    | [`Style`](/docs/styling.md)   |         |                              |
| `onSelectionEnter` | `Function` |  | Selection enter event callback  |
| `onSelectionLeave` | `Function` |  | Selection leave event callback  |

```typescript
type ResizeMode = 'contain' | 'cover' | 'stretch' | 'center' | 'repeat' | 'none';
```

#### Examples

Absolute path supported:

```javascript
<Image
  source="http://placekitten.com/400"
  resizeMode="contain"
  style={{
    height: 400,
    width: 400,
  }}
/>
```

Webpack image loader supported:

```javascript
import * as img from "./icon.png";
...
<Image
  source={img}
  resizeMode="contain"
  style={{
    height: 400,
    width: 400,
  }}
/>
```
