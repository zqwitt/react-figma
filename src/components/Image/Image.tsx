import * as React from 'react';
import { RectangleProps, Rectangle } from '../rectangle/Rectangle';
import { ResizeMode } from '../../styleTransformers/transformGeometryStyleProperties';
import { StyleSheet } from '../..';

export interface ImageProps extends RectangleProps {
    source: string;
    resizeMode?: ResizeMode;
}

export const Image: React.FC<ImageProps> = props => {
    const { style, source, resizeMode } = props;
    return (
        <Rectangle
            {...props}
            style={[StyleSheet.flatten(style), { backgroundImage: source, backgroundSize: resizeMode }]}
        />
    );
};
