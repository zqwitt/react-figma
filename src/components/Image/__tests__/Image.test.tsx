import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Image } from '../Image';

describe('<Image />', () => {
    it('Image with source and layout style properties', () => {
        const tree = renderer
            .create(
                <Image
                    source="./image.png"
                    style={{
                        width: 100,
                        height: 200
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Image with resizeMode prop', () => {
        const tree = renderer
            .create(
                <Image
                    source="./image.png"
                    resizeMode="stretch"
                    style={{
                        width: 100,
                        height: 200
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
