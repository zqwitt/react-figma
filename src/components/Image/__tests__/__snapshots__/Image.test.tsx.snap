// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> Image with resizeMode prop 1`] = `
<rectangle
  fills={Array []}
  height={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  resizeMode="stretch"
  source="./image.png"
  style={
    Array [
      Object {
        "height": 200,
        "width": 100,
      },
      Object {
        "backgroundImage": "./image.png",
        "backgroundSize": "stretch",
      },
    ]
  }
  width={100}
/>
`;

exports[`<Image /> Image with source and layout style properties 1`] = `
<rectangle
  fills={Array []}
  height={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  source="./image.png"
  style={
    Array [
      Object {
        "height": 200,
        "width": 100,
      },
      Object {
        "backgroundImage": "./image.png",
        "backgroundSize": undefined,
      },
    ]
  }
  width={100}
/>
`;
